compiler/rustc_lint/src/context/diagnostics/check_cfg.rs

```diff
@@ -42,6 +42,25 @@ fn to_check_cfg_arg(name: Symbol, value: Option<Symbol>, quotes: EscapeQuotes) -
     }
 }
 
+fn cargo_help_sub(inst: &impl Fn(EscapeQuotes) -> String) -> lints::UnexpectedCfgCargoHelp {
+    // We don't want to suggest the `build.rs` way to expected cfgs if we are already in a
+    // `build.rs`. We therefor do a best effort check to try to figure out if we are building a
+    // Cargo build script.
+    //
+    // Ideally we would check the crate name via `TyCtxt::crate_name`, but this isn't available
```

Comment:
With access to the session you can check for the `--crate-name` argument, right? `tcx.crate_name()` additionally looks for `#![crate_name]`, but as cargo unconditionally passes `--crate-name`, any `#![crate_name]` has to match `--crate-name`, so ignoring `#![crate_name]` here is fine.

Reply:
I didn't think of that, this could indeed use `sess.options.crate_name` and since this code is already behind [`was_invoked_from_cargo`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_session/utils/fn.was_invoked_from_cargo.html) we don't need to use the env verify if are actually coming from Cargo or not since it's already done.

Reply:
Done, thanks.